cmd/meroxa/root/apps/deploy.go

```diff
@@ -238,22 +242,24 @@ func (d *Deploy) getPlatformImage(ctx context.Context) (string, error) {
 
 func (d *Deploy) deployApp(ctx context.Context, imageName, gitSha, specVersion string) error {
 	d.logger.Infof(ctx, "Deploying application %q...", d.appName)
-	err := d.turbineCLI.Deploy(ctx, imageName, d.appName, gitSha, specVersion)
+	deploymentSpec, err := d.turbineCLI.Deploy(ctx, imageName, d.appName, gitSha, specVersion)
 	if err != nil {
 		return err
 	}
-
-	app, err := d.client.GetApplication(ctx, d.appName)
-	if err != nil {
-		return err
+	if specVersion != "" {
+		input := &meroxa.CreateDeploymentInput{
+			Application: meroxa.EntityIdentifier{Name: null.StringFrom(d.appName)},
+			GitSha:      gitSha,
+			SpecVersion: null.StringFrom(specVersion),
+			Spec:        null.StringFrom(deploymentSpec),
+		}
+		_, err = d.client.CreateDeployment(ctx, input)
```

Comment:
I thought this was going to be done from the API. 

Are we planning on making the API create the deployment in some point? I believe it should be more stable to leave the API the responsibility of doing that.

Reply:
I thought creating the deployment which carries the spec from the CLI was the way? Since this is what carries the IR?

Reply:
I don't understand the comment, Raul.  Could you rephrase?  What exactly did you think was going to be done in the API?  Did you think the application endpoint would create a populated deployment?  The Application endpoint will create an empty pipeline in v2, and the v1 application endpoint backfills an empty deployment for already-populated applications.

Reply:
this is the result of the second RFC.  this is what we do if we make sure the application exists before the deployment.  we create deployments and not applications after the application exists.